src/mongo_odm/nvp.hpp

```diff
@@ -74,17 +159,593 @@ struct Nvp {
     constexpr Nvp(T Base::*t, const char* name) : t(t), name(name) {
     }
 
+    /**
+     * Assignment operator that creaptes an update expression to set the field to a given value.
+     */
+    constexpr UpdateExpr<Nvp<Base, T>, no_opt_type> operator=(const no_opt_type& val) const {
+        return {*this, val, "$set"};
+    }
+
+    virtual std::string get_name() const {
```

Comment:
Also, should this actually be virtual? Or is this just for the final concrete type and crtp magic in NvpCRTP?